encoding/src/adapters/jpeg.rs

```diff
@@ -1,15 +1,54 @@
 //! Support for JPG image decoding.
 
-use super::MissingAttributeSnafu;
-use crate::adapters::{DecodeResult, PixelDataObject, PixelRWAdapter};
+use super::{ MissingAttributeSnafu, EncodeOptions };
+use crate::adapters::{EncodeResult, DecodeResult, PixelDataObject, PixelRWAdapter};
 use jpeg_decoder::Decoder;
 use snafu::{whatever, OptionExt, ResultExt};
 use std::io::Cursor;
+use jpeg_encoder::{Encoder, ColorType};
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
 pub struct JPEGAdapter;
 
 impl PixelRWAdapter for JPEGAdapter {
+
+    /// Encode DICOM image data into jpeg encoding
+    fn encode(
+            &self,
+            src: &dyn PixelDataObject,
+            options: EncodeOptions,
+            dst: &mut Vec<u8>,
+        ) -> EncodeResult<()> {
+
+            let cols = src.cols().unwrap();
```

Comment:
We must not `unwrap` here. Passing an object without pixel data would be enough for the whole thread to panic, with little opportunity for a graceful recovery.

For the lack of more specific error variants, it should be possible to do this (requires `use snafu::ResultExt`):

```suggestion
            let cols = src.cols()
                .whatever_context("Missing image DICOM attribute Columns");
```

The equivalent change should be done for all other `unwrap()` calls.

Reply:
This does not seem to work, unless I'm missing something, because there is an unsatisfied trait bound for the EncoderError struct, `FromString` I think from snafu.

I could instead return an EncodeError::CustomEncodeError instead (and maybe a MissingAttributeSnafu in the future, if it's appropriate).

Reply:
So you're right, some attributes were missing in the `CustomEncodeError` variant, so it still doesn't work with `whatever_context`. With #251 this should work more smoothly. So we can either try to get this implementation to work under the current limitations or we can base this contribution on #251. Now that v0.5.3 is out, it may be a good time to start pushing all breaking changes in for an eventual v0.6.0.

Reply:
It seems to be working now. I will push these changes after I finish the unit tests.